pyquil/tests/test_quantum_computer.py

```diff
@@ -583,3 +584,26 @@ def test_noisy(forest):
     qc = get_qc('1q-qvm', noisy=True)
     result = qc.run_and_measure(p, trials=10000)
     assert result[0].mean() < 1.0
+
+
+def test_orthogonal_array():
+    def bit_array_to_int(bit_array):
+        output = 0
+        for bit in bit_array:
+            output = (output << 1) | bit
+        return output
+
+    def check_random_columns(oa, strength):
+        num_q = oa.shape[1]
+        num_cols = min(num_q, strength)
+        column_idxs = random.sample(range(num_q), num_cols)
+        occurences = {entry: 0 for entry in range(2 ** num_cols)}
+        for row in oa[:, column_idxs]:
+            occurences[bit_array_to_int(row)] += 1
+        assert all([count == occurences[0] for count in occurences.values()])
+
+    for strength in [0, 1, 2, 3]:
+        num_q = random.randint(1, 32)
```

Comment:
For completeness' sake, we might just do this for every integer `[1, 32]`. I don't imagine this will slow the tests significantly, and we will also be confident over this interval rather than accidentally missing some `num_q` for which this test fails.

Reply:
Good point. It's fast, so I bumped up the max too.